docs/dev/go-style-guide.md

```diff
@@ -117,3 +118,200 @@ sdkerrors.Wrapf(
   <value 2>
 )
 ```
+
+## Common Mistakes
+
+This is a compilation of some of the common mistakes we see in the repo that should be avoided.
+
+---
+Keep receiver names short [Details here](https://google.github.io/styleguide/go/decisions#receiver-names)
+
+```go
+// bad
+func (chain *TestChain) NextBlock() {
+    res, err := chain.App.FinalizeBlock(&abci.RequestFinalizeBlock{
+        Height:             chain.ProposedHeader.Height,
+        Time:               chain.ProposedHeader.GetTime(),
+        NextValidatorsHash: chain.NextVals.Hash(),
+    })
+    require.NoError(chain.TB, err)
+    chain.commitBlock(res)
+}
+```
+
+```go
+// good
+func (c *TestChain) NextBlock() {
+    // Ommitted
+```
+
+---
+**Naked returns**
+
+Only acceptable in a small function. Once it's a medium-sized function, be explicit with your returned values. Similarly, do not name result parameters just because it enables you to use naked returns. Clarity is always more important than saving a few lines in your function.
+
+---
+**Struct defined outside of the package**
+
+Must have fields specified. [Referance](https://google.github.io/styleguide/go/decisions#field-names)
+
+```go
+// Good:
+r := csv.Reader{
+  Comma: ',',
+  Comment: '#',
+  FieldsPerRecord: 4,
+}
+```
+
+```go
+// Bad:
+r := csv.Reader{',', '#', 4, false, false, false, false}
+```
+
+---
+**Cuddled braces**
+
+[Dropping whitespace between braces](https://google.github.io/styleguide/go/decisions#cuddled-braces) for slice and array literals is only permitted when both of the following are true.
+  
+- The indentation matches
+- The inner values are also literals or proto builders (i.e. not a variable or other expression)
+
+```go
+// Good
+good := []*Type{
+    first, // Can't be cuddled
+    {Field: "second"},
+}
+```
+
+```go
+// Good:
+good := []*Type{{ // Cuddled correctly
+    Field: "value",
+}, {
+    Field: "value",
+}}
+```
+
+---
+**Function and method calls should not be separated based solely on line length**
```

Comment:
This is probably a bit of a controversial one for some ;)
I am personally good with either option, but happy to do this if laid out clearly in the google style guide. That said, this item should probably have a little more context and link back to the source.

There is also some inconsistency with the items as some are linked back and some are not. Would be good to have consistency in context and links